### PR TITLE
Update deprecated tuple type call's in cfunction

### PR DESCRIPTION
**Pull request by musm**
cc @yuyichao 

also can we run nanosolider here? 

partially address https://github.com/JuliaLang/julia/issues/23020
Does the old form need a depwarn? (someone else will have to take care of that)

### CODE REVIEW

**Comment by KristofferC** [comment]
@nanosoldier `runbenchmarks(ALL, vs = ":master")`

File: base/libuv.jl [hunk 1]
```diff
@@ -86,17 +86,17 @@ function process_events(block::Bool)
...
-    global uv_jl_sendcb        = cfunction(uv_sendcb, Void, (Ptr{Void}, Cint))
-    global uv_jl_return_spawn  = cfunction(uv_return_spawn, Void, (Ptr{Void}, Int64, Int32))
-    global uv_jl_asynccb       = cfunction(uv_asynccb, Void, (Ptr{Void},))
-    global uv_jl_timercb       = cfunction(uv_timercb, Void, (Ptr{Void},))
+    global uv_jl_alloc_buf     = cfunction(uv_alloc_buf, Void, Tuple{Ptr{Void}, Csize_t, Ptr{Void}})
+    global uv_jl_readcb        = cfunction(uv_readcb, Void, Tuple{Ptr{Void}, Cssize_t, Ptr{Void}})
+    global uv_jl_connectioncb  = cfunction(uv_connectioncb, Void, Tuple{Ptr{Void}, Cint})
+    global uv_jl_connectcb     = cfunction(uv_connectcb, Void, Tuple{Ptr{Void}, Cint})
+    global uv_jl_writecb_task  = cfunction(uv_writecb_task, Void, Tuple{Ptr{Void}, Cint})
+    global uv_jl_getaddrinfocb = cfunction(uv_getaddrinfocb, Void, Tuple{Ptr{Void},Cint,Ptr{Void}})
```
Comment (musm) [inline base/libuv.jl]: note to self for later, add white space between tuple types

Comment (fredrikekre) [inline base/libuv.jl]: Types within `{}` are typically not separated with spaces across base though?

Comment (musm) [inline base/libuv.jl]: True for the parametric syntax. For  `ccall`,  `invoke` and `cfunction` they all mostly have spaces between the types for readability it seems.  Also this is the only line here where the tuple type does not have a space between the types, so I thought I would make it consistent with the rest.

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/23046#issuecomment-318923101) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/0d5ec8b151978e175ae2b3867288375b5fa0d32d/3fb7502_vs_1ce3c8e/report.md). cc @ararslan

**Comment by musm** [comment]
Benchmarks look great. Not surprised since the old form has unnecessary allocations and is about 800x slower in simple test cases.

**Comment by martinholters** [comment]
Where does the one memory regression come from?

**Comment by KristofferC** [comment]
Nice find!

**Comment by musm** [comment]
Looks like noise, can rerun nanosoldier to confirm?

**Comment by fredrikekre** [comment]
@nanosoldier `runbenchmarks("rand", vs = ":master")`

**Comment by fredrikekre** [comment]
Woops, that should have been `"random"`.
@nanosoldier `runbenchmarks("random", vs = ":master")`

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/23046#issuecomment-319107436) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/be642f35fd1ee7941c971aff70c6c5cca538cadb/95a3a55_vs_09371a5/report.md). cc @ararslan

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/23046#issuecomment-319108366) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/34d4950d4b6f6d9da213095fe18130494ebb081a/95a3a55_vs_09371a5/report.md). cc @ararslan

**Comment by musm** [comment]
@fredrikekre  why "random" ?

**Comment by musm** [comment]
There seem to be a couple of timeouts, I am retriggering CI, but I believe several other PRs also are having this issue.

**Review by JeffBezanson — APPROVED** [review]

File: base/c.jl [hunk 2]
```diff
@@ -4,6 +4,23 @@
...
 import Core.Intrinsics: cglobal, bitcast
 
+"""
+    cfunction(function::Function, ReturnType::Type, ArgumentTypes::Type)
+
+Generate C-callable function pointer from Julia function. Type annotation of the return
+value in the callback function is a must for situations where Julia cannot infer the return
+type automatically.
+
+#Examples
```
Comment (tkelman) [inline base/c.jl]: `# Examples`

**Pull request merged**